joblib/executor.py

```diff
@@ -8,64 +8,99 @@
-                                      initializer=initializer,
-                                      initargs=initargs, env=env)
-    # If executor doesn't have a _temp_folder, it means it is a new executor
-    # and the reducers have not been used. Else, the previous reducers are used
-    # and we should not change this attribute.
-    if not hasattr(_executor, "_temp_folder"):
-        _executor._temp_folder = temp_folder
-    return _executor
-
-
-class _TestingMemmappingExecutor(TemporaryResourcesManagerMixin):
+def get_memmapping_executor(n_jobs, **kwargs):
+    return MemmappingExecutor.get_memmapping_executor(n_jobs, **kwargs)
+
+
+class MemmappingExecutor(_ReusablePoolExecutor):
+
+    @classmethod
+    def get_memmapping_executor(cls, n_jobs, timeout=300, initializer=None,
+                                initargs=(), env=None, temp_folder=None,
+                                context_id=None, **backend_args):
+        """Factory for ReusableExecutor with automatic memmapping for large numpy
+        arrays.
+        """
+        global _executor_args
+        # Check if we can reuse the executor here instead of deferring the test
+        # to loky as the reducers are objects that changes at each call.
+        executor_args = backend_args.copy()
+        executor_args.update(env if env else {})
+        executor_args.update(dict(
+            timeout=timeout, initializer=initializer, initargs=initargs))
+        reuse = _executor_args is None or _executor_args == executor_args
+        _executor_args = executor_args
+
+        manager = TemporaryResourcesManager(
+            temp_folder,
+            context_id=context_id
+        )
+
+        # reducers access the temporary folder in which to store temporary
+        # pickles through a call to manager.resolve_temp_folder_name. resolving
+        # the folder name dynamically is useful to use different folders across
+        # calls of a same reusable executor
+        job_reducers, result_reducers = get_memmapping_reducers(
+            unlink_on_gc_collect=True,
+            temp_folder_resolver=manager.resolve_temp_folder_name,
+            **backend_args)
+        _executor, executor_is_reused = super().get_reusable_executor(
+            n_jobs, job_reducers=job_reducers, result_reducers=result_reducers,
+            reuse=reuse, timeout=timeout, initializer=initializer,
+            initargs=initargs, env=env
+        )
+
+        if not executor_is_reused:
+            # if _executor is new, the previously created manager will used by
+            # the reducer to resolve temporary folder names. Otherwise, we
+            # musn't patch it, because the reducers will use the manager
+            # instance created by an older `get_memmaping_exeuctor` call.
+            _executor._temp_folder_manager = manager
+
+        return _executor
+
+    def terminate(self, kill_workers=False):
+        self.shutdown(kill_workers=kill_workers)
```

Comment:
Isn't the default `shutdown` with `wait=True`?  In this case, the children should all be collected, whatever the value of `kill_workers` no?

Reply:
Yes. Does something bother you?

Reply:
Then I don't get why you don't clean-up the files in both cases (in the next lines).
Sorry for putting the comment in a somewhat hard to read place!

Reply:
We **do** clean up the file in both cases, we just use a different method depending on if the workers are killed or not. Does this need an extra comment in the `else` close?

Reply:
indeed, I actually I misread.